### ISSUE TITLE
is it possible to test on random online photo?

### ISSUE THREAD

**Issue by datlife** [SummitKwan/transparent_latent_gan#14]
First of all, Great work! Thank you for sharing the code

I wonder if it is possible to test on custom online photo, e.g. using my photo on the kaggle notebook.

Thanks,

**Comment by SummitKwan** [comment]
Hi, thank you for asking this.  That is exactly my next step.  I have two methods in mind that I can try, but have not got time to implement.  Stay tuned.

**Comment by jackylee1** [comment]
me have the same question.wish to see your update.that's quite great work

**Comment by pixieDoug** [comment]
Looking for this as well!  How do you think you'll implement? Really great work!

**Comment by SummitKwan** [comment]
@pixieDoug Thank you! Basically, there are two strategies: 1) using gradient descent optimization to recover latent space https://arxiv.org/abs/1702.04782. 2) train a encoding network (CNN) to transform image to latent space.  I think either will work.

**Comment by pixieDoug** [comment]
I'd assume writing the gd optimizer is probably the easiest to implement and test results, no?  Think you'll have time to write some code? :)  I think testing with your own image and seeing the results would be cool to everyone.  And btw, congrats on making HN homepage. That's some very nice attention I'm sure.

**Comment by pixieDoug** [comment]
@SummitKwan The gd algorithm you referenced in the arXiv paper....do you know if there is a git repo? Would like to see if I can reproduce the results from their paper and maybe merge it here if successful.

**Comment by SummitKwan** [comment]
@pixieDoug Thank you so much for this and I really appreciate it.  One implementation is here: https://github.com/simoroma/RecoverGANlatentVector.  

However, the implementation of this gd is not the difficult part, the actual challenge here is to get the clean GAN computational graph from the Nvidia's implementation, which is currently wrapped in complicated classes (for a good reason of the progressively growing training procedure).  Let me know if you want to discuss a little bit more before devoting your time in it and feel free to reach my from LinkedIn message.  Again, thank you for your effort!

**Comment by pixieDoug** [comment]
@SummitKwan So sorry for not responding sooner! Unfortunately I've been tied-up on another DL project.  I would love to contribute but sounds like it might be past my knowledge of GANs. Still, I'd like to see what I can do to help push the project forward!  I'll look you up on LinkedIn and you can fill-me-in on the classes when you have a moment!

**Comment by abi98213** [comment]
Any Update, to test it on any image. I tried to generate latent vector from image but did not succeed. So, please tell me how should I approach this problem.

**Comment by chronos-ctrl** [comment]
Hi, any update on this ?

**Comment by tanmoyc-sph** [comment]
any update on this?..or any advise on the steps to achieve this ?
